branchLayout/api/src/main/java/com/virtuslab/branchlayout/api/BranchLayout.java

```diff
@@ -29,6 +30,28 @@ private static List<IBranchLayoutEntry> collectEntriesRecursively(IBranchLayoutE
     return entryByName.get(branchName).getOrNull();
   }
 
+  @Override
+  public @Nullable IBranchLayoutEntry findNextEntry(String branchName) {
+    val entriesOrderedList = rootEntries.flatMap(BranchLayout::collectEntriesRecursively).map(IBranchLayoutEntry::getName);
+    val currentIndex = entriesOrderedList.indexOf(branchName);
+    if (currentIndex > -1 && currentIndex + 1 < entriesOrderedList.length()) {
+      @LTLengthOf("entriesOrderedList") int nextIndex = currentIndex + 1;
+      return findEntryByName(entriesOrderedList.get(nextIndex));
+    }
+    return null;
+  }
+
+  @Override
+  public @Nullable IBranchLayoutEntry findPreviousEntry(String branchName) {
+    val entriesOrderedList = rootEntries.flatMap(BranchLayout::collectEntriesRecursively).map(IBranchLayoutEntry::getName);
+    val currentIndex = entriesOrderedList.indexOf(branchName);
+    val previousIndex = currentIndex - 1;
+    if (previousIndex > -1 && previousIndex < /* added for preventing a type error */entriesOrderedList.length()) {
```

Comment:
Maybe add `@LTLengthOf("entriesOrderedList")` to `previousIndex`, as in the above method?

Reply:
I tried that, it is not allowed syntactically, for some strange reason.

Reply:
sorry, the checker we have on the server, says that `currentIndex - 1` cannot be cast to that. Locally it compiles.

Reply:
Note that locally `IndexChecker` isn't executed by default... does the code compile locally even with `-PrunAllCheckers`?